src/net.cpp

```diff
@@ -1811,12 +1811,76 @@ void ThreadOpenConnections()
 
     // Initiate network connections
     int64_t nStart = GetTime();
-    while (true) {
+    unsigned int nDisconnects = 0;
+    while (true)
+    {
         ProcessOneShot();
 
         MilliSleep(500);
 
-        CSemaphoreGrant grant(*semOutbound);
+        // Only connect out to one peer per network group (/16 for IPv4).
+        // Do this here so we don't have to critsect vNodes inside mapAddresses critsect.
+        // And also must do this before the semaphore grant so that we don't have to block
+        // if the grants are all taken and we want to disconnect a node in the event that
+        // we don't have enough connections to XTHIN capable nodes yet.
+        int nOutbound = 0;
+        int nThinBlockCapable = 0;
+        set<vector<unsigned char> > setConnected;
+        CNode* ptemp = nullptr;
+        bool fDisconnected = false;
+        {
+            LOCK(cs_vNodes);
+            BOOST_FOREACH (CNode* pnode, vNodes)
+            {
+                if (pnode->fOutbound) // only count outgoing connections.
```

Comment:
I think it's fine the way it is for couple of reasons but if anyone strongly disagrees I'll change it:

1) I think it make the network stronger to have 4 outgoing because then we'll pretty much have 4 outs and 4 ins as everyone upgrades.  But other than that, what should the best count be?  I don't know...But I think think it works well as it is.

2)  Also, this is code for connecting outbound nodes, I think it's a little confusing to mix that with Inbound, although, it's not that big a deal.  So again, if someone else also feels strongly about it , i'll change it.

As for the logic of counting nThinBlockCapable, it works fine the way it is...not sure what you're getting at there?  We're only looking at outbound nodes here...